framework/src/engine/consensus/certificate_generation/commit_pool.ts

```diff
@@ -308,6 +312,7 @@ export class CommitPool {
 				maxHeightCertified + 1,
 			);
 			nextHeight = Math.min(heightNextBFTParameters - 1, maxHeightPrecommitted);
+			nextHeight = Math.max(nextHeight, this._minCertifyHeight - 1);
```

Comment:
Why do we do it differently here? Couldn't we add

`heightNextBFTParameters = Math.max(heightNextBFTParameters, this._minCertifyHeight);`

after line 313 instead for consistency?

Reply:
Assuming the `minCertrifyHeight` is one of the nextBFTParameters, i thought it's make sense because in L314, it is taking the min of `heightNextBFTParameters - 1, maxHeightPrecommitted`

Reply:
not sure I understand. The `nextHeight` calculated there should still be valid, I thought the problem was selecting the correct height of the 'first' BFT parameters.